indico/modules/events/contributions/controllers.py

```diff
@@ -16,42 +16,27 @@
 
 from __future__ import unicode_literals
 
-from flask import flash, request, jsonify, render_template
-from sqlalchemy.orm import joinedload
+
+from flask import flash, request, jsonify, redirect, session, render_template
 from werkzeug.exceptions import BadRequest
 
 from indico.modules.events.contributions.models.contributions import Contribution
 from indico.modules.events.contributions.operations import create_contribution, update_contribution, delete_contribution
 from indico.modules.events.contributions.forms import ContributionForm, ContributionProtectionForm
+from indico.modules.events.contributions.util import ContributionReporter
 from indico.modules.events.contributions.views import WPManageContributions
 from indico.modules.events.management.controllers import RHContributionPersonListMixin
 from indico.modules.events.models.events import Event
 from indico.modules.events.util import update_object_principals
 from indico.util.i18n import _, ngettext
 from indico.util.string import to_unicode
-from indico.web.flask.templating import get_template_module
+from indico.web.flask.util import url_for
 from indico.web.forms.base import FormDefaults
 from indico.web.util import jsonify_data, jsonify_form, jsonify_template
 from MaKaC.webinterface.rh.base import RH
 from MaKaC.webinterface.rh.conferenceModif import RHConferenceModifBase
 
-
-def _get_contribution_list_args(event):
-    timetable_entry_strategy = joinedload('timetable_entry')
-    timetable_entry_strategy.lazyload('*')
-    contribs = (event.contributions
-                .filter_by(is_deleted=False)
-                .order_by(Contribution.friendly_id)
-                .options(timetable_entry_strategy)
-                .all())
-    sessions = [{'id': s.id, 'title': s.title, 'colors': s.colors} for s in event.sessions.filter_by(is_deleted=False)]
-    tracks = [{'id': int(t.id), 'title': to_unicode(t.getTitle())} for t in event.as_legacy.getTrackList()]
-    return {'contribs': contribs, 'sessions': sessions, 'tracks': tracks}
-
-
-def _render_contribution_list(event):
-    tpl = get_template_module('events/contributions/management/_contribution_list.html')
-    return tpl.render_contribution_list(event=event, **_get_contribution_list_args(event))
+reporter = ContributionReporter
```

Comment:
Too damn long! It will cause a few line breaks and I think using the `reporter` instead of `ContributionReporter` makes the code more beautiful. Me no like it... Let's ask the others! @OmeGak @mic4ael @florv @pferreir ?